modules/api/src/main/scala/com/github/scoquelin/arugula/commands/RedisHashAsyncCommands.scala

```diff
@@ -1,19 +1,165 @@
 package com.github.scoquelin.arugula.commands
 
+import scala.collection.immutable.ListMap
 import scala.concurrent.Future
 
+import com.github.scoquelin.arugula.commands.RedisKeyAsyncCommands.ScanCursor
+
 /**
  * Asynchronous commands for manipulating/querying Hashes (key/value pairs)
  *
  * @tparam K The key type
  * @tparam V The value type
  */
 trait RedisHashAsyncCommands[K, V] {
+  /**
+   * Delete one or more hash fields
+   * @param key The key
+   * @param fields The fields to delete
+   * @return The number of fields that were removed from the hash
+   */
   def hDel(key: K, fields: K*): Future[Long]
+
+  /**
+   * Determine if a hash field exists
+   * @param key The key
+   * @param field The field
+   * @return True if the field exists, false otherwise
+   */
+  def hExists(key: K, field: K): Future[Boolean]
+
+  /**
+   * Get the value of a hash field
+   * @param key The key
+   * @param field The field
+   * @return The value of the field, or None if the field does not exist
+   */
   def hGet(key: K, field: K): Future[Option[V]]
+
+  /**
+   * get all the keys in a hash
+   * @param key The key
+   * @return A list of all the keys in the hash
+   */
+  def hKeys(key: K): Future[List[K]]
+
+  /**
+   * Get the number of fields in a hash
+   * @param key The key
+   * @return The number of fields in the hash
+   */
+  def hLen(key: K): Future[Long]
+
+  /**
+   * Get the values of all the fields in a hash
+   * @param key The key
+   * @return A list of all the values in the hash
+   */
   def hGetAll(key: K): Future[Map[K, V]]
+
+  /**
+   * Set the value of a hash field
+   * @param key The key
+   * @param field The field
+   * @param value The value
+   * @return True if the field is new, false if it was updated
+   */
   def hSet(key: K, field: K, value: V): Future[Boolean]
+
+  /**
+   * Get the values of multiple hash fields
+   * @param key The key
+   * @param fields The fields
+   * @return A map of field -> value for each field that exists or None if the field does not exist
+   */
+  def hMGet(key: K, fields: K*): Future[ListMap[K, Option[V]]]
+
+  /**
+   * Get the values of multiple hash fields
+   * @param key The key
+   * @param fields The fields
+   * @return A map of field -> value for each field that exists or None if the field does not exist
+   */
+  def hMGet(key: K, fields: => List[K]): Future[Map[K, Option[V]]]
```

Comment:
Should we keep both methods since they seem to achieve the same operation ?

Could we do maybe `def hMGet(key: K, fields: K*): Future[Map[K, Option[V]]]` only so that it's more in line with other operations ?

Reply:
I  personally hate the `fields: K*` approach since it prevents adding additional parameters later. Since that is unlikely to happen though, yeah I can get rid of the `def hMGet(key: K, fields: => List[K])` signature.

Reply:
okay, I removed the `def hMGet(key: K, fields: => List[K])` signature and left the other one. That should resolve the open issues.